prog/dftb+/prg_dftb/main.F90

```diff
@@ -0,0 +1,5637 @@
+  use fifo
+  use slakocont
+  use linkedlist
+  use lapackroutines
+  use mdcommon
+  use mdintegrator
+  use tempprofile
+  implicit none
+  private
+
+  public :: runDftbPlus
+
+  ! Name of the human readable files
+
+  !> Tagged output files (machine readable)
+  character(*), parameter :: autotestTag = "autotest.tag"
+
+  !> Detailed user output
+  character(*), parameter :: userOut = "detailed.out"
+
+  !> band structure and filling information
+  character(*), parameter :: bandOut = "band.out"
+
+  !> File accumulating data during an MD run
+  character(*), parameter :: mdOut = "md.out"
+
+  !> Machine readable tagged output
+  character(*), parameter :: resultsTag = "results.tag"
+
+  !> Second derivative of the energy with respect to atomic positions
+  character(*), parameter :: hessianOut = "hessian.out"
+
+  !> O(N^2) density matrix creation
+  logical, parameter :: tDensON2 = .false.
+
+  !> Should further output be appended to detailed.out?
+  logical, parameter :: tAppendDetailedOut = .false.
+
+
+contains
+
+  !> The main DFTB program itself
+  subroutine runDftbPlus()
+    use initprogram
+
+    !> Square dense hamiltonian storage for cases with k-points
+    complex(dp), allocatable :: HSqrCplx(:,:,:,:)
+
+    !> Square dense overlap storage for cases with k-points
+    complex(dp), allocatable :: SSqrCplx(:,:)
+
+    !> Square dense hamiltonian storage
+    real(dp), allocatable :: HSqrReal(:,:,:)
+
+    !> Square dense overlap storage
+    real(dp), allocatable :: SSqrReal(:,:)
+
+    !> Eigenvalues
+    real(dp), allocatable :: eigen(:,:,:)
+
+    !> Sparse storage of density matrix
+    real(dp), allocatable :: rhoPrim(:,:)
+
+    !> Imaginary part of density matrix in sparse storage
+    real(dp), allocatable :: iRhoPrim(:,:)
+
+    !> Energy weighted density matrix
+    real(dp), allocatable :: ERhoPrim(:)
+
+    !> Non-SCC part of the hamiltonian in sparse storage
+    real(dp), allocatable :: h0(:)
+
+    !> electronic filling
+    real(dp), allocatable :: filling(:,:,:)
+
+    !> band structure energy
+    real(dp), allocatable :: Eband(:)
+
+    !> entropy of electrons at temperature T
+    real(dp), allocatable :: TS(:)
+
+    !> zero temperature electronic energy
+    real(dp), allocatable :: E0(:)
+
+    !> energy in previous scc cycles
+    real(dp) :: Eold
+
+    !> Total energy components
+    type(TEnergies) :: energy
+
+    !> Potentials for orbitals
+    type(TPotentials) :: potential
+
+    !> Energy derivative with respect to atomic positions
+    real(dp), allocatable :: totalDeriv(:,:)
+
+    !> Forces on any external charges
+    real(dp), allocatable :: chrgForces(:,:)
+
+    !> excited state force addition
+    real(dp), allocatable :: excitedDerivs(:,:)
+
+    !> Stress tensors for various contribution in periodic calculations
+    real(dp) :: totalStress(3,3), totalLatDeriv(3,3)
+
+    !> derivative of cell volume wrt to lattice vectors, needed for pV term
+    real(dp) :: derivCellVol(3,3)
```

Comment:
@Ben can you suggest a better name? I could come up with `derivPV` or `derivPresVol`, but both look somewhat ugly.